pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -230,6 +244,66 @@ func TestLDROnNodeShutdown(
 	VerifyCorrectness(t, setup, leftJobID, rightJobID, 5*time.Minute)
 }
 
+// TestLDROnNetworkPartition aims to see what happens when both clusters
+// are separated from one another by a network partition. This test will
+// aim to keep the workload going on both sides and wait for reconciliation
+// once the network partition has completed
+func TestLDROnNetworkPartition(
+	ctx context.Context, t test.Test, c cluster.Cluster, setup multiClusterSetup,
+) {
+
+	duration := 10 * time.Minute
+	if c.IsLocal() {
+		duration = 3 * time.Minute
+	}
+
+	kvWorkload := replicateKV{
+		readPercent:             0,
+		debugRunDuration:        duration,
+		maxBlockBytes:           1024,
+		initRows:                1000,
+		initWithSplitAndScatter: true}
+
+	leftJobID, rightJobID := setupLDR(ctx, t, c, setup, kvWorkload)
+
+	monitor := c.NewMonitor(ctx, setup.CRDBNodes())
+	monitor.Go(func(ctx context.Context) error {
+		return c.RunE(ctx, option.WithNodes(setup.workloadNode), kvWorkload.sourceRunCmd("system", setup.CRDBNodes()))
+	})
+
+	// Let workload run for a bit before we kill a node
+	time.Sleep(kvWorkload.debugRunDuration / 10)
+
+	// Partition the left and right clusters so they can still make progress relative to each other,
+	// but can't talk to the nodes on the other side of the partition.
+	// Normally, the blackholeFailer is accessed through the failer interface,
+	// at least in the failover tests. Because this test shouldn't use all the
+	// failer interface calls (e.g. Setup(), and Ready()), we use the
+	// blakholeFailer struct directly. In other words, in this test, we
+	// shouldn't treat the blackholeFailer as an abstracted api.
```

Comment:
[nit] I'm not sure this comment is buying us much.

However, it does point out to me that the failer has a log of other failure modes. Might be useful to follow up with a test that uses it?

Reply:
Copied from Michael's test haha. I can remove it